### ISSUE TITLE
Provide interfaces for options with JSDocs

### ISSUE THREAD

**Issue by Shinigami92** [faker-js/faker#1586]
### Clear and concise description of the problem

As a user I want to have JSDoc tooltips when hovering over option properties in my IDE so I can directly see what the parameter is aiming for.

### Suggested solution

Provide interfaces for e.g. ranges.
Maybe even extract types for every ModuleMethodOption so I as a user can see e.g. default values for each specific property.

### Additional context

https://github.com/faker-js/faker/pull/1486#discussion_r1028049577

---

Automatisating the VitePress generated docs is a non-goal for the first PR. Everything that broke after this change for the docs need to be handled in a separate PR. That is so we can focus on IDE UX support in the first place.

**Comment by ST-DDT** [comment]
Just repeating this here so it doesn't get lost: 

> Please note, that I will only accept it for the entire project, if there is a test that ensures that the text is the same as the param's and optionally the respective parts are generated automatically (When using inline types).

---

I think one of the reasons, why we don't use named types, is that IDEs don't show their actual content and you have to click on it to get all properties. (It may potentially also requires some extra work to work with typedoc's `@param` detection/filtering, not a blocker though). So we should use these named types for only a few select params/options i.e. `NumberOrRange`.

![grafik](https://user-images.githubusercontent.com/1579362/203118047-0ae6965b-29cf-4c46-b785-b9b0f93a3196.png)

**Comment by ST-DDT** [comment]
**Team decision**

We discussed adding jsdocs to each option param directly in addition to the `@param` ones (basically a copy of that)

````ts
int({
  /**
   * The min value.
   */
  min: number,
  /**
   * The max value.
   */
  max: number
})
````

We decided not to use explicit types (e.g. AlphaOptions), because

- some methods have and require different types+descriptions for their attributes (exclude digits vs alpha chars)
- `@param` will no longer be checked
- It reduces the readability of the source code

**Comment by xDivisionByZerox** [comment]
I will close this since this issue has been solved by #1644 as far as I can see. If this is not correct please reopen this issue.
